### PR TITLE
Add REACT_APP_API_URL environment var to webpack config

### PR DESCRIPTION
**Pull request by abechoe**
This PR updates the webpack config to load environment variables upon build.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,5 +17,9 @@
     "radium": "^0.24.1",
     "react": "^16.4.2",
     "react-dom": "^16.4.2"
+  },
+  "scripts": {
+    "dev": "webpack --env.REACT_APP_API_URL=http://localhost:3000 --config webpack.config.js",
```
Comment (yirmiyahu) [inline package.json]: Considering that webpack is run in the context of node, could we not employ `dotenv` here so as to load env variables from an ignored `.env`?

Comment (yirmiyahu) [reply package.json]: That said, I don't think we should directly specifying the env variables here. We should just be specifying which environment is running (developement, production, etc.) and have the webpack config determine the env variables from there.

File: webpack.config.js [hunk 2]
```diff
@@ -1,50 +1,62 @@
...
+        '.js',
+        '.jsx',
+      ],
+    },
+    plugins: [
+      new HtmlWebpackPlugin({
+        template: path.resolve('index.html'),
+        inject: true,
+      }),
+      new webpack.DefinePlugin(envKeys)
```
Comment (yirmiyahu) [inline webpack.config.js:53]: Just curious what this is about?

File: webpack.config.js [hunk 3]
```diff
@@ -1,50 +1,62 @@
...
-    contentBase: path.join(__dirname, 'dist'),
-    publicPath: 'http://localhost:9000/dist',
-    port: 9000,
-    hotOnly: true,
-  },
-}
\ No newline at end of file
+    devServer: {
+      contentBase: path.join(__dirname, 'dist'),
+      publicPath: 'http://localhost:9000/dist',
```
Comment (yirmiyahu) [inline webpack.config.js]: Nitpick but isn't `http://localhost:9000` implied by specifying `port` below this?

File: webpack.config.js (hunk 2 above)
Comment (abechoe) [reply webpack.config.js:53]: `DefinePlugin` allows you to create a global constant that's configured at compile time: https://webpack.js.org/plugins/define-plugin/

Comment (yirmiyahu) [reply webpack.config.js:53]: I see. Neat.

File: webpack.config.js (hunk 3 above)
Comment (abechoe) [reply webpack.config.js]: Yes. I can remove it.

File: webpack.config.js [hunk 4]
```diff
@@ -1,50 +1,62 @@
...
-    contentBase: path.join(__dirname, 'dist'),
-    publicPath: 'http://localhost:9000/dist',
-    port: 9000,
-    hotOnly: true,
-  },
-}
\ No newline at end of file
+    devServer: {
+      contentBase: path.join(__dirname, 'dist'),
+      publicPath: 'http://localhost/dist',
```
Comment (yirmiyahu) [inline webpack.config.js]: Oh, again, not to make big deal, but can't this just be liek `'/dist/'` or something?

**Review by yirmiyahu — APPROVED** [review]

**Pull request merged**